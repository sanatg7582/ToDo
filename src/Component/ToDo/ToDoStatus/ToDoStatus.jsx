import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";
/**
 * This function ToDo status button
 * @param {bool} todoStatus
 * @param {func} handleOnClick
 * @param {number} index
 * @return element
 */
const ToDoStatus = props => {
  const { todoStatus, handleOnClick, index } = props;
  return (
    <>
      {todoStatus ? (
        <Button
          buttonType="button"
          btnClass="success"
          handleOnClick={handleOnClick}
          index={index}
          status={true}
        >
          <FontAwesomeIcon icon={faCheck} />
        </Button>
      ) : (
        <Button
          buttonType="button"
          btnClass="danger"
          handleOnClick={handleOnClick}
          index={index}
          status={false}
        >
          <FontAwesomeIcon icon={faTimes} />
        </Button>
      )}
    </>
  );
};
// default props value
ToDoStatus.defaultProps = {
  todoStatus: true,
  handleOnClick: () => {},
  index: 0
};
// prop type validation
ToDoStatus.propTypes = {
  todoStatus: PropTypes.bool.isRequired,
  handleOnClick: PropTypes.func.isRequired,
  index: PropTypes.number
};
export default ToDoStatus;
