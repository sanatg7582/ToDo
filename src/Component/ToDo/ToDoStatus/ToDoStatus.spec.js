import React from "react";
import { mount } from "enzyme";
import ToDoStatus from "./ToDoStatus";
describe("Load <ToDoStatus />", () => {
  let wrapper = "";
  beforeAll(() => {
    const props = {
      todoStatus: true,
      handleOnClick: () => {},
      index: 0
    };
    wrapper = mount(<ToDoStatus {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("checking button length", () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });
  it("checking props ", () => {
    // todo status true
    expect(wrapper.find("button").hasClass("btn success")).toEqual(true);
    expect(wrapper.props().todoStatus).toEqual(true);
    // todo status false
    wrapper.setProps({ todoStatus: false });
    expect(wrapper.find("button").hasClass("btn danger")).toEqual(true);
    expect(wrapper.props().todoStatus).toEqual(false);
  });

  it("checking button click", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("button").hasClass("btn danger")).toEqual(true);
  });

  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
