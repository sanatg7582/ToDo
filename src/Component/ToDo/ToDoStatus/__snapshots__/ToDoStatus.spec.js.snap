// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <ToDoStatus /> SnapShot test 1`] = `ReactWrapper {}`;
